acir/src/circuit/directives.rs

```diff
@@ -239,15 +247,26 @@ impl Directive {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-// If values are compile time and/or known during
-// evaluation, we can form an output string during ACIR generation.
-// Otherwise, we must store witnesses whose values will
-// be fetched during the PWG stage.
-pub enum LogInfo {
+/// This info is used when solving the initial witness
+/// If values are compile time and/or known during
+/// evaluation, we can form an output string during ACIR generation.
+/// Otherwise, we must store witnesses whose values will
+/// be fetched during the PWG stage.
+pub enum LogOutputInfo {
     FinalizedOutput(String),
     WitnessOutput(Vec<Witness>),
 }
 
+pub struct SolvedLog {
+    pub is_trace: bool,
+    pub output_info: SolvedLogOutputInfo,
+}
+
+pub enum SolvedLogOutputInfo {
+    FinalizedOutput(String),
+    WitnessValues(Vec<FieldElement>),
```

Comment:
If the log is solved, why do you keep the values and not just have a string?

Reply:
I didn't want to leave any decoding of the field elements. Transforming `Vec<FieldElement>` into a string right away would also require the extra logic that currently exists in the ACVM where we check whether we have a singular FieldElement or [multiple](https://github.com/noir-lang/acvm/blob/0b305b7bc18cc9f1569b4ddb09ac50a48ecef5f9/acvm/src/pwg/directives.rs#L172) (as the string looks different based on whether we have a singular field element or an array).